core/src/schema/content/array.rs

```diff
@@ -1,15 +1,138 @@
+            }
+
+            fn visit_map<A>(self, mut access: A) -> Result<Self::Value, A::Error>
+            where
+                A: de::MapAccess<'de>,
+            {
+                let mut length = None;
+                let mut content = None;
+
+                while let Some(key) = access.next_key()? {
+                    match key {
+                        Field::Length => {
+                            if length.is_some() {
+                                return Err(de::Error::duplicate_field("length"));
+                            }
+
+                            length = Some(access.next_value()?);
+                        }
+                        Field::Content => {
+                            if content.is_some() {
+                                return Err(de::Error::duplicate_field("content"));
+                            }
+
+                            content = Some(access.next_value()?);
+                        }
+                    }
+                }
+
+                let length = length.ok_or_else(|| de::Error::missing_field("length"))?;
+                let content = content.ok_or_else(|| de::Error::missing_field("content"))?;
+
+                match length {
+                    Content::Number(NumberContent::U64(number_content::U64::Range(r))) => {
+                        if r.high.is_none() {
+                            return Err(A::Error::custom(
+                                "missing high value for array length range"
+                            ));
+                        }
+                    }
+                    // Default for ranges
+                    Content::Number(NumberContent::U32(number_content::U32::Range(r))) => {
+                        if r.high.is_none() {
+                            return Err(A::Error::custom(
+                                "missing high value for array length range"
+                            ));
+                        }
+                    }
+                    Content::Number(NumberContent::U64(_)) => {},
+                    // Default for negative numbers
+                    Content::Number(NumberContent::I64(number_content::I64::Constant(n))) => {
+                        is_positive(n).map_err(A::Error::custom)?
+                    }
+                    Content::Number(NumberContent::I32(number_content::I32::Range(r))) => {
+                        if r.high.is_none() {
+                            return Err(A::Error::custom(
+                                "missing high value for array length range"
+                            ));
+                        }
+
+                        is_positive(
+                            r.low .unwrap_or_default() .into(),
+                        ).map_err(A::Error::custom)?
+                    },
+                    Content::SameAs(_) => {},
+                    Content::Null(_) => return Err(de::Error::custom("array length is missing. Try adding '\"length\": 5' to the array type")),
```

Comment:
I think the message for 'null' should probably match the `"length": [number] where...` format of the message on the line below, just avoid confusing users.